src/components/accordion/accordion.js

```diff
@@ -0,0 +1,268 @@
+  this.setExpanded(!expanded, $section)
+
+  // Store the state in sessionStorage when a change is triggered
+  this.storeState($section)
+}
+
+// When Open/Close All toggled, set and store state
+Accordion.prototype.onOpenOrCloseAllToggle = function () {
+  var $module = this
+  var $sections = this.$sections
+
+  var nowExpanded = !this.checkIfAllSectionsOpen()
+
+  nodeListForEach($sections, function ($section) {
+    $module.setExpanded(nowExpanded, $section)
+    // Store the state in sessionStorage when a change is triggered
+    $module.storeState($section)
+  })
+
+  $module.updateOpenAllButton(nowExpanded)
+}
+
+// Set section attributes when opened/closed
+Accordion.prototype.setExpanded = function (expanded, $section) {
+  var $button = $section.querySelector('.' + this.sectionButtonClass)
+  $button.setAttribute('aria-expanded', expanded)
+
+  if (expanded) {
+    $section.classList.add(this.sectionExpandedClass)
+  } else {
+    $section.classList.remove(this.sectionExpandedClass)
+  }
+
+  // See if "Open all" button text should be updated
+  var areAllSectionsOpen = this.checkIfAllSectionsOpen()
+  this.updateOpenAllButton(areAllSectionsOpen)
+}
+
+// Get state of section
+Accordion.prototype.isExpanded = function ($section) {
+  return $section.classList.contains(this.sectionExpandedClass)
+}
+
+// Check if all sections are open
+Accordion.prototype.checkIfAllSectionsOpen = function () {
+  // Get a count of all the Accordion sections
+  var sectionsCount = this.$sections.length
+  // Get a count of all Accordion sections that are expanded
+  var expandedSectionCount = this.$module.querySelectorAll('.' + this.sectionExpandedClass).length
+  var areAllSectionsOpen = sectionsCount === expandedSectionCount
+
+  return areAllSectionsOpen
+}
+
+// Update "Open all" button
+Accordion.prototype.updateOpenAllButton = function (expanded) {
+  var newButtonText = expanded ? 'Close all' : 'Open all'
+  newButtonText += '<span class="govuk-visually-hidden"> sections</span>'
+  this.$openAllButton.setAttribute('aria-expanded', expanded)
+  this.$openAllButton.innerHTML = newButtonText
+}
+
+// Check for `window.sessionStorage`, and that it actually works.
+var helper = {
+  checkForSessionStorage: function () {
+    var testString = 'this is the test string'
+    var result
+    try {
+      window.sessionStorage.setItem(testString, testString)
+      result = window.sessionStorage.getItem(testString) === testString.toString()
+      window.sessionStorage.removeItem(testString)
+      return result
+    } catch (exception) {
+      if ((typeof console === 'undefined' || typeof console.log === 'undefined')) {
+        console.log('Notice: sessionStorage not available.')
+      }
+    }
+  }
+}
+
+// Set the state of the accordions in sessionStorage
+Accordion.prototype.storeState = function ($section) {
+  if (this.browserSupportsSessionStorage) {
+    // We need a unique way of identifying each content in the accordion. Since
```

Comment:
Is `window.sessionStorage` scoped to the current page, or is it across domain?

If it's shared across a domain, and the same IDs are reused across multiple pages (e.g. `#section1`) then open sections will persist across different pages (e.g. the first section might be open when you navigate to a page you haven't seen before) which I think would be unexpected?

Reply:
They are persisted across a domain. We discussed this [here](https://github.com/alphagov/govuk-frontend/pull/958#issuecomment-431290759) and decided that we should advise in guidance that the service team will need to use unique-across-service IDs for their accordion items - we didn't feel that the onus of generating these with JS should be with the component code. 

It's a really good point since the guidance doesn't currently address this - this is quite a "code-y" point so I'm not sure whether the Design System component page is the best place for this guidance, maybe it should live in the component README 🤔

Reply:
👍

Reply:
I've added an explanation to the [component options](https://github.com/alphagov/govuk-frontend/blob/bfb58c04ed1a7f9c368c75da16f1e8a06509d1d7/src/components/accordion/accordion.yaml#L5).